api/src/services/itis-service.test.ts

```diff
@@ -182,6 +182,65 @@ describe('ItisService', () => {
     });
   });
 
+  describe('getHierarchyForTSNs', async () => {
+    it('returns array of hierarchy objects for tsns', async () => {
+      const mockTsns = [1, 2];
+      const mockTsnHierarchies = [
+        { tsn: mockTsns[0], hierarchyTSN: ['$3$2$1$'] },
+        { tsn: mockTsns[1], hierarchyTSN: ['$3$2$'] }
+      ];
+      const mockAxiosResponse = {
+        data: {
+          response: {
+            docs: mockTsnHierarchies
+          }
+        }
+      };
+
+      const getItisSolrTsnHierarchyUrlStub = sinon
+        .stub(ItisService.prototype, 'getItisSolrTsnHierarchyUrl')
+        .returns('url');
+
+      const axiosStub = sinon.stub(axios, 'get').resolves(mockAxiosResponse);
+
+      const itisService = new ItisService();
+
+      const response = await itisService.getHierarchyForTSNs(mockTsns);
+
+      expect(getItisSolrTsnHierarchyUrlStub).to.have.been.calledWith(mockTsns);
+      expect(axiosStub).to.have.been.calledWith('url');
+      expect(response).to.eql([
+        {
+          tsn: mockTsns[0],
+          hierarchy: [3, 2, 1]
+        },
+        {
+          tsn: mockTsns[1],
+          hierarchy: [3, 2]
+        }
+      ]);
+    });
+
+    it('catches and re-throws an error', async () => {
+      const mockTsns = [1, 2];
+      sinon.stub(axios, 'get').rejects(new Error('a test error'));
+
+      const itisService = new ItisService();
+      const getItisSolrTsnHierarchyUrlStub = sinon
+        .stub(ItisService.prototype, 'getItisSolrTsnHierarchyUrl')
+        .resolves('url');
+
+      try {
+        await itisService.getHierarchyForTSNs(mockTsns);
+
+        expect.fail();
+      } catch (error) {
+        expect((error as ApiGeneralError).message).to.equal('a test error');
+        expect(getItisSolrTsnHierarchyUrlStub).to.have.been.calledWith(mockTsns);
+      }
+    });
```

Comment:
This test isn't really accurate. The function doesn't catch any errors, it would just let them bubble up (which is totally fine). I would just remove this test.
```suggestion
```

Reply:
Cool feature if you haven't seen it yet: can suggest edits in the PR and they become commitable changes (only really good for small/obvious changes though).